documentation/jetty/modules/code/examples/src/main/java/org/eclipse/jetty/docs/programming/ArchitectureDocs.java

```diff
@@ -75,4 +79,60 @@ public void virtualVirtualThreads()
         HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP(clientConnector));
         // end::virtualVirtual[]
     }
+
+    public static class EitherTask implements Invocable.Task
+    {
+        private final TryExecutor executor;
+        private final Deque<Invocable.Task> subTasks = new ConcurrentLinkedDeque<>();
+
+        public EitherTask(TryExecutor executor)
+        {
+            this.executor = executor;
+        }
+
+        @Override
+        public InvocationType getInvocationType()
+        {
+            return InvocationType.EITHER;
+        }
+
+        public void offer(Task task)
+        {
+            subTasks.add(task);
+        }
+
+        @Override
+        public void run()
+        {
+            if (Invocable.isNonBlockingInvocation())
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING, EITHER -> subTask.run();
+                        case BLOCKING -> executor.execute(subTask);
+                    }
+                }
+            }
+            else
+            {
+                for (Task subTask = subTasks.pollFirst(); subTask != null; subTask = subTasks.pollFirst())
+                {
+                    switch (Invocable.getInvocationType(subTask))
+                    {
+                        case NON_BLOCKING -> subTask.run();
+                        case EITHER -> Invocable.invokeNonBlocking(subTask);
```

Comment:
I don't agree.  We always try to consume a task with the current thread if possible.  We also do not deliberately do a parallel fan out, rather we simply go parallel to avoid head of line blocking.   

This is an important distinction, because it is often better to do 8 things serially on one CPU, rather than try to do them in parallel on 8 CPUs,  especially when they are all from one connection and your 7 other CPUs are dealing with other connections.

Our AES does not do parallel fan out if it can avoid it.

Reply:
The current AES does a `tryExecute()` for EITHER tasks, see https://github.com/jetty/jetty.project/blob/jetty-12.0.14/jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/strategy/AdaptiveExecutionStrategy.java#L338.

Reply:
Actually - I'm wrong.... we do try to treat an EITHER task as a BLOCKING task and only go to invoke if there is no reserved thread.   I'll change to match.

Reply:
@gregw note that this has the great side effect of parallelization.

If our AES did not do that, we would have less efficient HTTP/3 because we would not able to parallelize different QUIC connections -- they would be processed sequentially.

However, this is a nice feature of the implementation; the definition/javadocs of EITHER does not require mentioning it.